app/controllers/games_controller.rb

```diff
@@ -0,0 +1,38 @@
+class GamesController < ApplicationController
+  before_action :authenticate_user!, only: [:new, :create, :update]
+
+  def new
+    @game = Game.new
+  end
+
+  def create
+    @game = Game.create(game_params.merge(white_player_id: current_user.id))
+    if @game.persisted?
+      redirect_to game_path(@game)
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def show
+    begin 
+      @game = Game.find(params[:id])
+    rescue ActiveRecord::RecordNotFound
+      if @game.nil?
```

Comment:
As I mentioned in my earlier comment, there's no need to check if `@game == nil`. Since we're now inside the `rescue` block, you can be confident that an `ActiveRecord::RecordNotFound` exception was thrown by `find` and that it's appropriate to generate a 404 using `render`.